pkg/reconciler/contour/resources/kingress.go

```diff
@@ -51,6 +51,12 @@ func MakeEndpointProbeIngress(ctx context.Context, ing *v1alpha1.Ingress, previo
 
 	// Reverse engineer our previous state from the prior generation's HTTP Proxy resources.
 	for _, proxy := range previousState {
+		// Skip probe when status is not valid. It happens when the previous revision was garbage collected.
+		// see: https://github.com/knative/serving/issues/9582
+		if proxy.Status.CurrentStatus != "valid" {
+			continue
+		}
+
```

Comment:
I don't think we log the proxy status anywhere (and really don't key off of this anywhere else).  I'd definitely log here (at INFO), but it'd be useful to get logging around the proxy status in generate (at DEBUG).  That would probably help track this and other issues down via logstream.

/unhold

Reply:
Thank you. I added Info log. But sorry I didn't understand about where is the best place for `it'd be useful to get logging around the proxy status in generate (at DEBUG)`.

Reply:
Sorry, I meant that we should put that Info log statement alongside the `continue`.

For the Debug log statement, maybe we should put it in the create/update loop (after it fetches a proxy)?

Reply:
Alright. I updated.
The create loop already had the debug log https://github.com/knative-sandbox/net-contour/blob/600a323c5b92d260dcd637058bea1fd3c7f9e15a/pkg/reconciler/contour/contour.go#L211 so just added to update loop.